ios/RCTMarkdownUtils.mm

```diff
@@ -47,13 +48,17 @@ - (NSAttributedString *)parseMarkdown:(nullable NSAttributedString *)input
   JSValue *result = [function callWithArguments:@[inputString]];
   NSArray *ranges = [result toArray];
 
-  NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithString:inputString attributes:_backedTextInputView.defaultTextAttributes];
+  NSMutableAttributedString *attributedString = [input mutableCopy];
   [attributedString beginEditing];
 
+  NSMutableDictionary<NSAttributedStringKey, id> *attributes = [_backedTextInputView.defaultTextAttributes mutableCopy];
+  [attributes removeObjectForKey:RCTTextAttributesTagAttributeName];
+  [attributedString addAttributes:attributes range:NSMakeRange(0, attributedString.length)];
+
   // If the attributed string ends with underlined text, blurring the single-line input imprints the underline style across the whole string.
   // It looks like a bug in iOS, as there is no underline style to be found in the attributed string, especially after formatting.
-  // This is a workaround that applies the NSUnderlineStyleNone to the string before iterating over ranges which resolves this problem.
-  [attributedString addAttribute:NSUnderlineStyleAttributeName value:[NSNumber numberWithInteger:NSUnderlineStyleNone] range:NSMakeRange(0, attributedString.length)];
+  // The workaround is to remove NSUnderlineStyleAttributeName attribute for whole string before iterating over ranges.
+  [attributedString removeAttribute:NSUnderlineStyleAttributeName range:NSMakeRange(0, attributedString.length)];
```

Comment:
This previous version of this workaround breaks the diacritics context:

https://github.com/Expensify/react-native-live-markdown/assets/20516055/24a508fa-4bff-48e7-b1b7-224c05a932fb

If we remove the workaround, the diacritics context works correctly but the link underline persists:

https://github.com/Expensify/react-native-live-markdown/assets/20516055/1a9e8772-f190-4168-a773-0e952f7b57af

If we modify the workaround so it uses `removeAttribute`, it works correctly:

https://github.com/Expensify/react-native-live-markdown/assets/20516055/9e568d37-9070-4e08-aa8f-96a2c0ad702d

Reply:
Can you check on the single line input as well? I'm testing on it, and the current version of the underline workaroud doesn't work for me when `multiline={false}`
<details>
<summary>Reproduction steps</summary>

1. set `multiline` prop of `MarkdownTextInput` to `false`
2. Write any link inside text input
3. press enter to submit - notice that it blurs the input
4. focus the input and start typing - notice that underline is applied to whole input


https://github.com/Expensify/react-native-live-markdown/assets/61146594/6b204b68-9c73-4e0d-8db4-6237f5d9d6be


</details>